src/ol/control/ScaleLine.js

```diff
@@ -274,11 +275,13 @@ class ScaleLine extends Control {
       }
       ++i;
     }
-
     let html;
     if (this.scaleBar_) {
       html = this.createScaleBar(width, count, suffix);
     } else {
+      if (count % 1 !== 0) {
```

Comment:
You could play this save and replace the if-block with something that also works for numbers where you don't want decimal places:
```js
html = count.toFixed(decimalCount < 1 ? -decimalCount : 0) + ' ' + suffix;
```